packages/devtools-local-toolbox/src/clients/chrome.js

```diff
@@ -38,51 +40,70 @@ function connectClient() {
   const deferred = defer();
 
   if(!getValue("chrome.debug")) {
-    return Promise.resolve(createTabs([]))
+    return Promise.resolve(createTabs([], {}))
   }
 
-  const webSocketPort = getValue("chrome.webSocketPort");
-  const url = `http://localhost:${webSocketPort}/json/list`;
-  networkRequest(url).then(res => {
-    deferred.resolve(createTabs(JSON.parse(res.content)))
-  }).catch(() => deferred.reject());
+  const port = getValue("chrome.port");
+  function fetch(http, options, callback) {
+    const {host, path, port} = options;
+    const url = `http://${host}:${port}${path}`;
+    return networkRequest(url).then(res => callback(null, res.content))
+  }
```

Comment:
@cyrus-and I'm not if passing in a custom [fetcher](https://github.com/cyrus-and/chrome-remote-interface/compare/master...jasonLaster:fetcher?expand=1) is the best idea or just a dirty hack.

The problem I'm trying solve is this. The debugger runs in a web context so fetching `localhost:9222/json/list` will be a CORS error. What we do instead is use `networkRequest` which builds a url `localhost:8000/?get=localhost:9222/json/list` to do the request for us. 

What I'm doing here is passing in `fetch` as my go-to fetcher with CRI, but it feels bad. Have you seen others try to do this before?